src/components/PrivateRoute/PrivateRoute.jsx

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+const PrivateRoute = ({ children }) => {
+	const currentUSer = useSelector((state) => state.user);
+
+	return (
+		<>{currentUSer.role === 'admin' ? children : <Navigate to='/courses' />}</>
```

Comment:
Note: Private router HOC is for checking user is logged in or not. 
scenario -> 
1) if auth token is present show the requested route
2) if auth token is not present redirect to login page